web/src/engine/websites/Delitoon.ts

```diff
@@ -0,0 +1,174 @@
+}
+
+//Result from /auth/refresh : to refresh token
+type APIRefreshToken = {
+    result: APIToken
+}
+
+//Auth token entity
+type APIToken = {
+    accessToken: {
+        token: string,
+        expiredAt: number
+    },
+    refreshToken: {
+        token: string
+    }
+}
+
+@Common.ImageAjax()
+export default class extends DecoratableMangaScraper {
+    protected BalconyID: string = undefined;
+    protected Timezone: string = 'Europe/Paris';
+    protected Platform: string = 'WEB';
+    protected Token: APIToken = undefined;
+
+    public constructor(id = 'delitoon', label = 'Delitoon', url = 'https://www.delitoon.com', balconyID = 'DELITOON_COM', tags = [Tags.Media.Manhwa, Tags.Language.French, Tags.Source.Official]) {
+        super(id, label, url, ...tags);
+        this.BalconyID = balconyID;
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/detail/[^/]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const mangaid = new URL(url).href.match(/\/detail\/([^/]+)/)[1];
+        const req = new URL(`/api/balcony-api-v2/contents/${mangaid}`, this.URI);
+        req.searchParams.set('isNotLoginAdult', 'true');
+        const { data } = await FetchJSON<APIResult<APIManga>>(this.CreateRequest(req));
+        return new Manga(this, provider, mangaid, data.title.trim());
+    }
+
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const url = new URL('/api/balcony-api-v2/contents/search', this.URI);
+        url.search = new URLSearchParams({
+            searchText: '',
+            isCheckDevice: 'true',
+            isIncludeAdult: 'true',
+            contentsThumbnailType: 'MAIN'
+        }).toString();
+        const { data } = await FetchJSON<APIResult<APIManga[]>>(this.CreateRequest(url));
+        return data.map(element => new Manga(this, provider, element.alias, element.title.trim()));
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        await this.GetToken();
+        const url = new URL(`/api/balcony-api-v2/contents/${manga.Identifier}`, this.URI);
+        url.searchParams.set('isNotLoginAdult', 'true');
+        const { data } = await FetchJSON<APIResult<APIManga>>(this.CreateRequest(url));
+        return data.episodes.map(element => {
+            let title = element.title.trim();
+            title += element.subTitle ? ' : ' + element.subTitle.trim() : '';
+            return new Chapter(this, manga, element.alias, title);
+        });
+    }
+
+    public override async FetchPages(chapter: Chapter): Promise<Page[]> {
+        await this.GetToken();
+        const url = new URL(`/api/balcony-api-v2/contents/${chapter.Parent.Identifier}/${chapter.Identifier}`, this.URI);
+        url.searchParams.set('isNotLoginAdult', 'true');
```

Comment:
Is this parameter required when fetching pages?

Reply:
![image](https://github.com/manga-download/haruneko/assets/10671518/cfe70aa5-dfd3-44fe-aec3-c9c7bf6ee082)

Reply:
It shows that /viewer/ is now a valid endpoint for, perhaps all websites. Gotta correct that (old still working)